PochtaBot.py

```diff
@@ -0,0 +1,274 @@
+                        answer = MESSAGE_INVALID_SHIPMENT
+
+            else:
+                # user entry has invalid format
+                answer = MESSAGE_INVALID_SHIPMENT_FORMAT
+
+    except Exception as e:
+        print("Exception processing user entry", e)
+        answer = MESSAGE_GENERAL_ERROR
+        if DEBUG:
+            raise e
+
+    if not answer:
+        print("Empty answer - logical failure")
+        answer = MESSAGE_GENERAL_ERROR
+
+    markup = draw_buttons(message.chat.id)
+    bot.send_message(message.chat.id, answer, reply_markup=markup)
+
+
+def schedule_checker():
+    while True:
+        schedule.run_pending()
+        sleep(1)  # unload CPU between schedule runs
+
+
+# this procedure checks if there is update to shipment registered in DB
+def automated_notification_procedure():
+    try:
+        with db_session:
+            for chat_id in auto_notified_chats:
+                non_delivered_shipments = select(
+                    shipment for shipment in Shipment if shipment.chat == chat_id and shipment.last_event != 2
+                )
+
+                for non_delivered_shipment in non_delivered_shipments:
+
+                    shipment_xml = RussianPostAPI.get_shipment_data(
+                        non_delivered_shipment.barcode, CFG_POSTAL_API_KEY, CFG_POSTAL_API_PASS
+                    )
+                    shipment_info = ShipmentInfoParser.parse_xml(shipment_xml)
+
+                    if shipment_info.events[-1][0] != non_delivered_shipment.last_event or \
+                            shipment_info.events[-1][4] != non_delivered_shipment.last_event_result:
+                        answer = MESSAGE_STATUS_UPDATE + str(non_delivered_shipment.barcode) + "\n" + get_shipment_description(shipment_info, 0)
+                        non_delivered_shipment.last_event = shipment_info.events[-1][0]
+                        non_delivered_shipment.last_event_result = shipment_info.events[-1][4]
+
+                        # as a result of above code some items may become "delivered",
+                        # so we need to hide them so we have to redraw buttons
+                        markup = draw_buttons(chat_id)
+
+                        bot.send_message(chat_id, answer, reply_markup=markup)
+
+    except Exception as e:
+        print("Exception in Auto Notification Procedure", e)
+        if DEBUG:
+            raise e
+
+
+# Create the job with schedule
+schedule.every(CFG_AUTO_NOTIFICATION_INTERVAL).minutes.do(automated_notification_procedure)
+
+# Spin up a thread to run the schedule check so it doesn't block the bot.
+# Teacher please understand and forgive there is no thread safety implemented.
+Thread(target=schedule_checker).start()
+
+while True:
+    print("The bot is now running")
+    try:
+        bot.polling(none_stop=True, interval=0)
+
+    except Exception as e:
+        print("Message polling has restarted with the following reason:", e)
+        if DEBUG:
+            raise e
```

Comment:
No newline

Reply:
added a separation line and a empty line at the end of the file